fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -2579,6 +2582,12 @@ ACTOR Future<Void> reply(CommitBatchContext* self) {
 		                                     self->prevVersion,
 		                                     writtenTags),
 		    TaskPriority::ProxyMasterVersionReply));
+		if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST) {
+			ASSERT(self->loggingComplete.isReady());
+			ASSERT_WE_THINK(self->loggingComplete.get() >= self->commitVersion); // @todo remove later
```

Comment:
It turns out this assertion doesn't always hold true. Here's what "self->loggingComplete" gets set to (maybe right after the cluster start):

self->loggingComplete: 1, self->commitVersion: 427116

Is it to do with how tLogs capture the "durableKnownCommittedVersion" (which is what "TagPartitionedLogSystem::push()" returns) or is there a bug in how the returned version is getting computed here https://github.com/apple/foundationdb/blob/edae0db41bcce244059ec9c1fff7038c3cc26ad1/fdbserver/TagPartitionedLogSystem.actor.cpp#L632 when unicast is enabled? I will investigate further.

Reply:
The current code does not update `minKnownCommittedVersion` if the choose statement exits on condition `when(wait(pProxyCommitData->committedVersion.whenAtLeast(self->commitVersion + 1))) {}`. However, you are updating it unconditionally. Are there any implications to that?

Reply:
My understanding of this code might not be totally correct, but I think it is not getting updated here because there is another transaction (which updated "pProxyCommitData->committedVersion" to be greater than "self->commitVersion") would have updated this. Even if we want to update it here we wouldn't know what to update it to because we haven't received a reply from the LogSystem yet and so we don't know the "minKnownCommittedVersion".

Reply:
It is not getting updated unconditionally here because this condition https://github.com/apple/foundationdb/blob/5497488e411735beb1e52f3ee559cd17fa2258f8/fdbserver/CommitProxyServer.actor.cpp#L2569 holds true here.

Reply:
I ran a Joshua job by modifying the code to set "ProxyCommitData::minKnownCommittedVersion" here in all cases (that is, even with version vector disabled). No failures (job id: 20240717-072746-sre-f1265d524d69cf67).

Reply:
Regarding the assertion failure: Here are the sample values of "self->loggingComplete" (the version returned by "TagPartitionedLogSystem::push()") and "self->commitVersion" in "reply()" actor (from a simulation test run):

> LoggingComplete: 1, Commit Version: 427116
> LoggingComplete: 427116, Commit Version: 434831
> LoggingComplete: 427116, Commit Version: 437417
> LoggingComplete: 437417, Commit Version: 442819
> LoggingComplete: 442819, Commit Version: 462323
> LoggingComplete: 442819, Commit Version: 467325
> LoggingComplete: 467325, Commit Version: 488582
> LoggingComplete: 488582, Commit Version: 861065
> LoggingComplete: 861065, Commit Version: 963394
> LoggingComplete: 963394, Commit Version: 1450091
> LoggingComplete: 1450091, Commit Version: 1487528
> LoggingComplete: 1487528, Commit Version: 2526807
> LoggingComplete: 1487528, Commit Version: 2530482
> LoggingComplete: 2530482, Commit Version: 2537508
> LoggingComplete: 2537508, Commit Version: 2567811
> LoggingComplete: 2537508, Commit Version: 2571070
> LoggingComplete: 2571070, Commit Version: 2586617
> LoggingComplete: 2571070, Commit Version: 2591827
> LoggingComplete: 2591827, Commit Version: 2610342
> LoggingComplete: 2591827, Commit Version: 2618881

As we can see, the version returned by "TagPartitionedLogSystem::push()" (and hence the value of "ProxyCommitData::minKnownCommittedVersion") is always lagging behind "CommitBatchContext::commitVersion" and in most of the cases it is the value of the previous commit version.

Reply:
I think we need to understand "TagPartitionedLogSystem::push()" better in order to explain why it's returned version is below the commit version. But probably that is not a blocker for this PR.